src/main/java/com/icthh/xm/ms/entity/service/processor/FormSpecProcessor.java

```diff
@@ -0,0 +1,166 @@
+package com.icthh.xm.ms.entity.service.processor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.icthh.xm.ms.entity.domain.spec.FormSpec;
+import com.icthh.xm.ms.entity.domain.spec.XmEntitySpec;
+import com.icthh.xm.ms.entity.service.JsonListenerService;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+@Slf4j
+@Component
+public class FormSpecProcessor extends SpecProcessor {
+    private static final String REF_FORM_PATTERN = "#/xmEntityForm/**/*";
+    private static final String KEY_FORM_TEMPLATE = "#/xmEntityForm/{formKey}";
+    private static final int POSSIBLE_RECURSIVE_LIMIT = 10;
+    private final Map<String, Map<String, FormSpec>> formsByTenant;
+
+    public FormSpecProcessor(JsonListenerService jsonListenerService) {
+        super(jsonListenerService);
+        this.formsByTenant = new ConcurrentHashMap<>();
+    }
+
+    public void updateFormStateByTenant(String tenant, Map<String, Map<String, String>> typesByTenantByFile) {
+        var definitionEntitySpec = new LinkedHashMap<String, FormSpec>();
+        typesByTenantByFile.get(tenant).values().stream().map(this::toFormSpecsMap).forEach(definitionEntitySpec::putAll);
```

Comment:
is it possible to use collect() here instead of mutation external map definitionEntitySpec?

Reply:
@amedvedchuk From my point of view, it is not convenient in this case. Because toFormSpecsMap has the same logic as [toTypeSpecsMap](https://github.com/xm-online/xm-ms-entity/blob/ceb618613700397f17f1c4954e5a594d311f05d7/src/main/java/com/icthh/xm/ms/entity/service/XmEntitySpecService.java#L608) and returns already completed map which we need. If we use the collect method, it will still be wrapped in a collection and we need to get it back. As a result, we will have additional actions.

Reply:
@monassfaro 
hm... i was thinking about something like:
```
var definitionEntitySpec = typesByTenantByFile.get(tenant)
                .values()
                .stream()
                .map(this::toFormSpecsMap)
                .map(Map::entrySet)
                .flatMap(Collection::stream)
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
              ```
so here we can clearly see the shain of stream transformation and avoid external collection population. 
Why do you think it is inconvenient?

Reply:
@amedvedchuk oh, I see. There was some misunderstanding. Now it's seems clearly, thanks. Fixed.